[PetMatch-Bugfixes]

kosteekX3D opened the pull request:
Сделано:
1) Task #16174 -- [back] Поправить обработку ошибок
2) Bug #16345 -- Формат ошибки при создании пользователя с существующим email
3) Bug #16407 -- Неправильный response code при регистрации с несуществующим городом
4) Bug #16408 -- Неверная валидация максимальной длины имени
5) Bug #16409 -- Неверный код ответа при валидации почты при логине
6) Bug #16410 -- Неверный код ответа при попытке логина с несуществующим юзером
7) Bug #16348 -- Неверная валидация времени встреч при регистрации
8) Bug #16350 -- Невереный или отсутсвующий лимит на количество временных интервалов встречи при регистрации
9) Bug #16351 -- Неверная валидация ФИО
10) Bug #16349 -- Неверный или отсутсвующий лимит на количество контактов при регистрации
11) Bug #16468 -- Превышение максимального количества контактов проходит валидацию
12) Bug #16417 -- Имя пользователя: формат не соответствует требованиям
13) Bug #16418 -- Изменение данных пользователя: параметры локации не соответствуют требованиям
14) Bug #16465 -- Превышение максимального количества интервалов для связи проходит валидацию
15) Bug #16466 -- Пересекающиеся интервалы для связи проходят валидацию

kosteekX3D committed:
Bug #16350

- Issue fix: https://ai.nsu.ru/issues/16350

kosteekX3D committed:
Bug #16349

- Issue fix: https://ai.nsu.ru/issues/16349

kosteekX3D committed:
Bug #16351

- Issue fix: https://ai.nsu.ru/issues/16351

kosteekX3D committed:
- Issue fix: https://ai.nsu.ru/issues/16348
- Issue fix: https://ai.nsu.ru/issues/16345
- Issue fix: https://ai.nsu.ru/issues/16347
-Task #16174 done

kosteekX3D committed:
issues fixes

kosteekX3D committed:
Bug #16408

- Issue fix: https://ai.nsu.ru/issues/16408

kosteekX3D committed:
Bug #16410

- Issue fix: https://ai.nsu.ru/issues/16410

kosteekX3D committed:
Bugs #16410 + #16409

- Issue fix: https://ai.nsu.ru/issues/16409
- Issue fix: https://ai.nsu.ru/issues/16410

kosteekX3D committed:
Bug #16407

- Issue fix: https://ai.nsu.ru/issues/16407

kosteekX3D committed:
Merge branch 'develop' into PetMatch-Bugfixes

kosteekX3D committed:
added tests

Zertalian1 commented:
Какие номера багов? 
Обсуждали, как должны называться ветки уже не один раз, почему эта называется так

kosteekX3D committed:
.

kosteekX3D commented:
> Какие номера багов? Обсуждали, как должны называться ветки уже не один раз, почему эта называется так

Добавил описание, мердж конfликт убрал, остался правда вопрос, там в test/ был GlobalExceptionHandler, я его убрал и там используется наш главный GlobalExceptionHandler, хз норм ли это.

kosteekX3D committed:
multiple fixes

- Issue fix: https://ai.nsu.ru/issues/16468
- Issue fix: https://ai.nsu.ru/issues/16466
- Issue fix: https://ai.nsu.ru/issues/16465
- Issue fix: https://ai.nsu.ru/issues/16418

kosteekX3D committed:
Bug #16417

https://ai.nsu.ru/issues/16417

Zertalian1 commented on src/main/java/org/nsu/users/dto/responses/UserProfileResponse.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Nonnull
+    @Schema(description = "Пол пользователя", example = "M", allowableValues = {"M", "F"})
+    private final String gender;
```
а чего не енамом

kosteekX3D replied:
Решено.

Zertalian1 commented on src/main/java/org/nsu/users/services/UserProfileService.java:
```diff
@@ -29,19 +31,54 @@ public class UserProfileService {
+                .bondTime(bondTimeDtos)
+                .contactInfo(contactDtos)
+                .reviewStatus(user.getStatus().getName())
+                .reviewComment(null)
```
а почему null?

kosteekX3D replied:
Решено.

Zertalian1 commented on src/main/java/org/nsu/users/utils/ValidationPatterns.java:
```diff
@@ -3,8 +3,25 @@
+    /**
+     * Pattern matching time in HH:mm 24-hour format (00:00 - 23:59)
+     */
+    public static final String TIME_HH_MM = "^(?:[01]\\d|2[0-3]):[0-5]\\d$";
```
а чем формат TIME_FORMAT = "HH:mm" не устроил

kosteekX3D replied:
Решено.

Zertalian1 commented on src/main/java/org/nsu/users/validation/ValidBondTimeIntervalsValidator.java:
```diff
@@ -0,0 +1,123 @@
+import java.time.LocalTime;
+import java.util.List;
+
+public class ValidBondTimeIntervalsValidator implements ConstraintValidator<ValidBondTimeIntervals, List<?>> {
```
А не проще было сделать утилу с проверкой) Ты же проверяешь буквально для 2 запросов - создание и редактирование.

kosteekX3D replied:
Решено. (если правильно понял)

Zertalian1 commented on src/main/resources/db/changelog/changes/petMatch-15170.xml:
```diff
@@ -113,11 +113,11 @@
                 <constraints nullable="false"/>
             </column>
-            <column name="first_name" type="VARCHAR(30)">
+            <column name="first_name" type="VARCHAR(64)">
```
нельзя. Хочешь менять - меняй в отдельном чейндже. От таких правок, бэк на стенде не стартует

kosteekX3D replied:
Решено.

Zertalian1 commented on src/test/java/org/nsu/authorization/core/controllers/LoginControllerIT.java:
```diff
@@ -0,0 +1,141 @@
+                + "org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfiguration,"
+                + "org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration"
+})
+class LoginControllerIT {
```
Что за IT это тест, пиши Test

kosteekX3D replied:
Решено. Но там некоторые тесты были с IT - Integrаtion Test и подумал мб так же надо.

kosteekX3D committed:
fix

Andrvat commented:
@kosteekX3D реши конфликты

kosteekX3D committed:
conflicts after merge

kosteekX3D commented:
> @kosteekX3D реши конфликты

@Andrvat готово.
